packages/monitoring-system/data-processor/test/unit/firestore-schema.spec.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+import {describe, it} from 'mocha';
+import assert from 'assert';
+import {getPrimaryKey, FirestoreCollection} from '../../src/firestore-schema';
+
+describe('firestore-schema', () => {
+  describe('getPrimaryKey', () => {
+    it('returns the correct Bot document primary key', () => {
+      const document = {bot_name: 'foo'};
+      assert.equal(getPrimaryKey(document, FirestoreCollection.Bot), 'foo');
+    });
+    it('returns the correct Bot_Execution document primary key', () => {
```

Comment:
Is the plan that you'll ultimately write a variety of unit tests like this, for helpers related to database access, vs., integration tests against firestore?

Reply:
There are lots of unit tests [here](https://github.com/googleapis/repo-automation-bots/blob/master/packages/monitoring-system/data-processor/test/unit/data-processors/cloud-logs-data-processor.spec.ts) that test against a mock (in-memory) firestore which has been pretty robust so far.

Eventually I'll also be adding integration tests against an actual Firestore in the cloud, but the issue with that is it doesn't let me test error handling (I cannot force the cloud Firestore to throw an error). So that will mostly be happy-path testing